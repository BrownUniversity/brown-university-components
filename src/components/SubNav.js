import PropTypes from "prop-types";
import React from "react";
import Nav from "./Nav";

const SubNav = ({ children, ...restProps }) => (
  <nav {...restProps}>
    <Nav sub>{children}</Nav>
  </nav>
);

SubNav.propTypes = {
  children: PropTypes.node.isRequired,
};

SubNav.Item = Nav.Item;
SubNav.Link = Nav.Link;

export default SubNav;
