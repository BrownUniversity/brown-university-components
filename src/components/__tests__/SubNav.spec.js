import React from 'react';
import { render } from 'react-testing-library';

import SubNav from '../SubNav';

const renderSubNav = ({
  props = {},
  children = (
    <React.Fragment>
      <SubNav.Item>
        <SubNav.Link>Link</SubNav.Link>
      </SubNav.Item>
      <SubNav.Item>
        <SubNav.Link active>Active Link</SubNav.Link>
      </SubNav.Item>
      <SubNav.Item>
        <SubNav.Link disabled>Link</SubNav.Link>
      </SubNav.Item>
    </React.Fragment>
  )
} = {}) => {
  const rtlUtils = render(<SubNav {...props}>{children}</SubNav>);

  return {
    subNav: rtlUtils.container.firstChild,
    ...rtlUtils
  };
};

describe('SubNav', () => {
  it('should render default', () => {
    const { subNav } = renderSubNav();

    expect(subNav).toMatchSnapshot();
  });
});
