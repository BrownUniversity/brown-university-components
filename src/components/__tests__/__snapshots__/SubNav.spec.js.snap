// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubNav should render default 1`] = `
.c1 {
  display: list-item;
  padding: 8px 0 8px 0;
}

.c2 {
  background: transparent;
  border: none;
  display: block;
  font-family: Circular-Book,Arial,Helvetica,sans-serif;
  padding: 0;
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
  color: #C00404;
  cursor: pointer;
  font-size: 1.1em;
  font-weight: 400;
  line-height: 1.5;
  opacity: 1;
  pointer-events: auto;
  -webkit-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
}

.c2:hover {
  color: #000000;
}

.c3 {
  background: transparent;
  border: none;
  display: block;
  font-family: Circular-Book,Arial,Helvetica,sans-serif;
  padding: 0;
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
  color: #000000;
  cursor: pointer;
  font-size: 1.1em;
  font-weight: 400;
  line-height: 1.5;
  opacity: 1;
  pointer-events: auto;
  -webkit-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
}

.c3:hover {
  color: #000000;
}

.c4 {
  background: transparent;
  border: none;
  display: block;
  font-family: Circular-Book,Arial,Helvetica,sans-serif;
  padding: 0;
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
  color: #C00404;
  cursor: not-allowed;
  font-size: 1.1em;
  font-weight: 400;
  line-height: 1.5;
  opacity: 0.65;
  pointer-events: auto;
  -webkit-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
}

.c4:hover {
  color: #C00404;
}

.c0 {
  list-style: none;
  margin: 0;
  background-color: #F7F7F7;
  display: block;
  padding: 15px;
  width: 100%;
}

<nav>
  <ul
    class="c0"
  >
    <li
      class="c1"
    >
      <button
        class="c2"
        color="red"
      >
        Link
      </button>
    </li>
    <li
      class="c1"
    >
      <button
        class="c3"
        color="red"
      >
        Active Link
      </button>
    </li>
    <li
      class="c1"
    >
      <button
        class="c4"
        color="red"
        disabled=""
      >
        Link
      </button>
    </li>
  </ul>
</nav>
`;
